### ISSUE TITLE
Deploying stuck at Installing Appx Package...

### ISSUE THREAD

**Issue by sarkarstanmoy** [ms-iot/vscode#16]
I know the issue has been already raised but it was closed without giving any proper solution/explanation. Actually I have tried to start NodeScriptHost and in the output it is showing 
Activating NodeScriptHost_1.0.1.0_arm__dnsz84vs3g3zp
Application Started

But still not able to deploy any application neither Node Script Host is listed in application.

Can you please let me know the exact issue?

**Comment by IoTGirl** [comment]
Hi @sarkarstanmoy ,

The issue was closed because we could not reproduce the same experience with the information provided.  As asked at https://github.com/ms-iot/vscode/issues/14 would you like to provide the steps you took to see the problem?  When we follow the steps given we do not get the error.

The supposition was that the steps were not followed and potentially there is a setup/configuration error resulting in the error you are seeing but we are unable to reproduce it therefore have no answer for you.

Sincerely,
IoTGirl

**Comment by sarkarstanmoy** [comment]
Thanks for the reply. Can you point out the detailed steps which I am suppose to follow to get rid of this error? In that case I can compare and find the missing step.

**Comment by IoTGirl** [comment]
I am sorry if my answer was not clear. We do not know what steps you took to get that error so have no insight into which step you may have missed or what may be unique about your setup.

**Comment by sarkarstanmoy** [comment]
Thanks for the reply. I will look into it and let you know. However, if you have, by any chance, any url to share it will be great about Windows IoT setup using nodejs.

**Comment by paulmon** [comment]
Are you following the steps here: https://marketplace.visualstudio.com/items?itemName=ms-iot.windowsiot ?

**Comment by sarkarstanmoy** [comment]
Yes paulman I have followed the steps with no success. Just wonder do I need to install Microsoft node chakra to make it work.

**Comment by sarkarstanmoy** [comment]
Here is my device info:
Get Device Info:
Device=192.168.0.106
ComputerName=minwinpc
Language=en-us
OsEdition=IoTUAP
OsEditionId=123
OsVersion=15063.0.armfre.rs2_release.170317-1834
Platform=Raspberry Pi 2 Model B

Does it work for only Raspberry pi 3?

**Comment by sarkarstanmoy** [comment]
Hi ,
Today I downloaded the extension and follow the steps to build. I am able to successfully build the extension and started debugging.  While debugging I found  that the extension will download NodeScriptHost file from azure if it is missing and uploaded to Windows IoT. Since I have downloaded the source code and run it I am able to upload the node js file without any issue. 
```javascript
Run Remote Script:
Downloading resources:
d:\Development\WindowsIoT\vscode-master\ScriptHostExtension\arm\NodeScriptHost_1.0.1.0_ARM.appx
d:\Development\WindowsIoT\vscode-master\ScriptHostExtension\arm\NodeScriptHost_1.0.1.0_ARM.cer
d:\Development\WindowsIoT\vscode-master\ScriptHostExtension\arm\Microsoft.VCLibs.ARM.14.00.appx
Download complete
Installing Appx Package...
NodeScriptHost is not running.
Locating files to upload in workspace.
[50%] Uploading d:\Development\WindowsIoT\VSCode-IoT\index.js
[100%] Uploading d:\Development\WindowsIoT\VSCode-IoT\package.json
Upload Completed
Starting NodeScriptHost...
Done.
```
However if I downloaded from VS MarketPlace then I am facing the same issue mentioned above i.e deployment stuck at Installing Appx Package. Request you to kindly look into it.

**Comment by Glober777** [comment]
I am having the same issue and I can't figure out how to get NodeScriptHost pushed out the my RPi3. Can someone please provide the steps? Do I need to compile using the sources from the GIT repo or there's already a built package that I can grab?

I've executed all the steps as they're listed in the overview for the extension against a newly installed Windows IoT Core on my raspberry several times now with no success. Please help.
